internal/builders/gradle/action.yml

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Gradle builder
+inputs:
+  # BYOB-provided inputs
+  slsa-workflow-inputs:
+    # Inputs in JSON format.
+    description: 'All the onputs'
+    type: string
+    required: true
+  slsa-layout-file:
+    description: 'Location to store the layout content'
+    type: string
+    required: true
+  slsa-workflow-secret1:
+    description: 'secret1'
+    type: string
+    required: true
+  slsa-workflow-secret2:
+    description: 'secret2'
+    type: string
+    required: true
+
+on:
+  workflow_call:
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK
+      uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: ${{ fromJson(inputs.slsa-workflow-inputs).jdk-version }}
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2
+      with:
+        arguments: build -x test
+    - name: Put release artifacts in one directory
+      shell: bash
+      env:
+        SLSA_OUTPUTS_ARTIFACTS_FILE: ${{ inputs.slsa-layout-file }}
+        ARTIFACT_LIST: ${{ fromJson(inputs.slsa-workflow-inputs).artifact-list }}
+      run: |
+        find . -name "sigstore-java-*-javadoc.jar"
```

Comment:
nit: maybe separate this out into a shell script so that we can run shell linters and simplify the workflow?